recipes/qwt/all/conanfile.py

```diff
@@ -0,0 +1,119 @@
+        "opengl": [True, False],
+        "mathml": [True, False],
+        "designer": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "plot": True,
+        "widgets": True,
+        "opengl": True,
+        "designer": True,
+        "mathml": False,
+        "svg": False
+
+    }
+    generators = "qmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("qt/5.15.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def _patch_qwt_config_files(self):
+        # qwtconfig.pri
+        qwtconfig_path = os.path.join(self.source_folder, self._source_subfolder, "qwtconfig.pri")
+        qwtconfig = tools.load(qwtconfig_path)
+
+        qwtconfig = "CONFIG += conan_basic_setup\ninclude(../conanbuildinfo.pri)\n" + qwtconfig
+        qwtconfig += "QWT_CONFIG {}= QwtDll\n".format("+" if self.options.shared else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtPlot\n".format("+" if self.options.plot else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtWidgets\n".format("+" if self.options.widgets else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtSvg\n".format("+" if self.options.svg else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtOpenGL\n".format("+" if self.options.opengl else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtMathML\n".format("+" if self.options.mathml else "-")
+        qwtconfig += "QWT_CONFIG {}= QwtDesigner\n".format("+" if self.options.designer else "-")
+        tools.save(qwtconfig_path, qwtconfig)
+
+        # qwtbuild.pri
+        qwtbuild_path = os.path.join(self.source_folder, self._source_subfolder, "qwtbuild.pri")
+        qwtbuild = tools.load(qwtbuild_path)
+        # set build type
+        qwtbuild += "CONFIG -= debug_and_release\n"
+        qwtbuild += "CONFIG -= build_all\n"
+        qwtbuild += "CONFIG -= release\n"
+        qwtbuild += "CONFIG += {}\n".format("debug" if self.settings.build_type == "Debug" else "release")
+        if self.settings.build_type == "RelWithDebInfo":
+            qwtbuild += "CONFIG += force_debug_info\n"
+        tools.save(qwtbuild_path, qwtbuild)
+
+    def build(self):
+        self._patch_qwt_config_files()
+
+        if self.settings.compiler == "Visual Studio":
+            vcvars = tools.vcvars_command(self.settings)
+            self.run("{} && qmake {}".format(vcvars, self._source_subfolder), run_environment=True)
+            self.run("{} && nmake".format(vcvars))
+        else:
+            self.run("qmake {}".format(self._source_subfolder), run_environment=True)
+            self.run("make -j {}".format(tools.cpu_count()))
```

Comment:
Is it robust?

What about MinGW?
Cross compilation? (qt is not in build requirements, I guess qmake comes from qt recipe).

Reply:
yes you are right: qmake comes from qt recipe.
Can a recipe be both a requirement and a build_requirement at the same time ?

Reply:
yes, it works like a charm with 2 profiles. It also work with 1 profile, but it breaks quickly when versions are different in requirements and build requirements.

AFAIR, conan plans to change 1 profile so that it behaves like 2 profiles.

Until it's implemented, you can do:

```python
def build_requirements(self):
    if tools.cross_building(self.settings):
        self.requires("qt/5.15.2")
```

Reply:
qmake is always intended to run on the build target, so it should be fine without this change right ?

Reply:
Conan v2 will always work as if there were two profiles defined. Here in CCI we are now adding our first cross-building scenario (we are fighting with M1) and the plan is to start using two profiles for all the builds (for native ones) at some point in time. We need to check that they work as expected, specially regarding the computation of package IDs.

Reply:
I don't think it's required to have things like MinGW and cross-compilation in the first iteration.
having recipe what works only for native builds is still better than having no recipes, and it would probably satisfy ~90% of users.
things like MinGW and cross-building could be added on demand.